### PR TITLE
Ability to create a tt:// link to share a channel

### PR DESCRIPTION
**Pull request by CoBC**

### CODE REVIEW

File: Client/qtTeamTalk/mainwindow.cpp [hunk 1]
```diff
@@ -5093,6 +5096,42 @@ void MainWindow::slotChannelsDeleteFile(bool /*checked =false */)
...
+    QInputDialog inputDialog;
+    inputDialog.setOkButtonText(tr("&Ok"));
+    inputDialog.setCancelButtonText(tr("&Cancel"));
+    inputDialog.setInputMode(QInputDialog::TextInput);
+    inputDialog.setTextValue(m_host.username);
+    inputDialog.setWindowTitle(tr("Share channel"));
+    inputDialog.setLabelText(tr("Type username to use to share this channel:"));
+    ok = inputDialog.exec();
+    if (inputDialog.textValue().size()>0)
+        link += QString("&username=%5").arg(inputDialog.textValue());
```
Comment (bear101) [inline Client/qtTeamTalk/mainwindow.cpp:5113]: You need to URL-encode the username.

File: Client/qtTeamTalk/mainwindow.cpp [hunk 2]
```diff
@@ -5093,6 +5096,42 @@ void MainWindow::slotChannelsDeleteFile(bool /*checked =false */)
...
+    if (ok)
+    {
+        inputDialog.setTextEchoMode(QLineEdit::Password);
+        inputDialog.setTextValue(m_host.password);
+        inputDialog.setWindowTitle(tr("Share channel"));
+        inputDialog.setLabelText(tr("Type password of this user account:"));
+        ok = inputDialog.exec();
+        if (ok && inputDialog.textValue().size()>0)
+            link += QString("&password=%6").arg(inputDialog.textValue());
```
Comment (bear101) [inline Client/qtTeamTalk/mainwindow.cpp:5122]: You need to URL-encode the password.

File: Client/qtTeamTalk/mainwindow.cpp [hunk 3]
```diff
@@ -5093,6 +5096,42 @@ void MainWindow::slotChannelsDeleteFile(bool /*checked =false */)
...
+    }
+    if (TT_GetMyChannelID(ttInst) > 0)
+    {
+        link += QString("&channel=%7").arg(m_mychannel.szName);
+        if (m_mychannel.bPassword)
+            link += QString("&chanpasswd=%8").arg(m_mychannel.szPassword);
```
Comment (bear101) [inline Client/qtTeamTalk/mainwindow.cpp:5128]: You need to URL-encode the channel password.

File: Client/qtTeamTalk/mainwindow.cpp [hunk 4]
```diff
@@ -5093,6 +5096,42 @@ void MainWindow::slotChannelsDeleteFile(bool /*checked =false */)
...
+        inputDialog.setTextValue(m_host.password);
+        inputDialog.setWindowTitle(tr("Share channel"));
+        inputDialog.setLabelText(tr("Type password of this user account:"));
+        ok = inputDialog.exec();
+        if (ok && inputDialog.textValue().size()>0)
+            link += QString("&password=%6").arg(inputDialog.textValue());
+    }
+    if (TT_GetMyChannelID(ttInst) > 0)
+    {
+        link += QString("&channel=%7").arg(m_mychannel.szName);
```
Comment (bear101) [inline Client/qtTeamTalk/mainwindow.cpp:5126]: You need to URL-encode the channel name.

File: Client/qtTeamTalk/mainwindow.ui [hunk 5]
```diff
@@ -2070,6 +2071,17 @@
     <string>Ctrl+S</string>
    </property>
   </action>
+  <action name="actionShareChannel">
+   <property name="checkable">
+    <bool>false</bool>
+   </property>
+   <property name="text">
+    <string>Create a tt:// link to this Channel</string>
```
Comment (bear101) [inline Client/qtTeamTalk/mainwindow.ui:2079]: Menu items use capital letter. Instead use:
Create a tt:// URL to This Channel

**Comment by bear101** [comment]
I guess it's this issue: #692

File: Client/qtTeamTalk/mainwindow.cpp (hunk 1 above)
Comment (CoBC) [reply Client/qtTeamTalk/mainwindow.cpp:5113]: Have you an example for this? I guess it's the QUrl::toPercentEncoding function?

**Comment by bear101** [comment]
Should we close #692 ?

**Comment by CoBC** [comment]
Yes I think. It has already be closed when you merged.

**Pull request merged**